<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Next Generation Sequencing Toolkit &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/css/adctheme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <meta name="generator" content="Sphinx/ADC Theme by Mirounga www.mirounga.fr"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="BitByBit  documentation" href="index.html" />
    <link rel="next" title="RNASeq Tutorial" href="rnaseq.html" />
    <link rel="prev" title="Whole Genomes Toolkit" href="wholegenomes.html" />
    <link media="only screen and (max-device-width: 480px)" href="_static/css/mobile.css" type="text/css" rel="stylesheet" /> 
  </head><body>
<div id="docstitle">
    <p>BitByBit  documentation</p>
</div>
<div id="header">
    <div id="title"><h1>Next Generation Sequencing Toolkit</h1></div>
    <ul id="headerButtons">
        <li id="toc_button"><div class="headerButton"><a href="#">Table of Contents</a></div></li>
        <li id="page_buttons">
            <div class="headerButton"><a href="genindex.html" title="General Index" accesskey="I">index</a></div>
            <div class="headerButton"><a href="rnaseq.html" title="RNASeq Tutorial" accesskey="N">next</a></div>
            <div class="headerButton"><a href="wholegenomes.html" title="Whole Genomes Toolkit" accesskey="P">previous</a></div>
        </li>
    </ul>
</div>

<div id="sphinxsidebar">
  <div class="sphinxsidebarwrapper">
        <ul><li class="toctree-l1"><a href="index.html">Main Page</a></li></ul>
        <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="rvassocanalysis.html"><strong>Rare Variants Association Analysis</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="snparray.html"><strong>SNP Array Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="wholegenomes.html"><strong>Whole Genomes Toolkit</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Next Generation Sequencing Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseq.html"><strong>RNASeq Tutorial</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html"><strong>Mapping using Salmon</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling1.html"><strong>Samtools Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling2.html"><strong>GATK Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="vcfrandom.html"><strong>VCF Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="SV.html"><strong>Structural Variation</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseqvariants.html"><strong>Variant Calling for RNASeq Data</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="somatics.html"><strong>Somatic Mutations from RNAseq</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="lncRNA.html"><strong>Long nonCoding RNA Prediction</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseqexpressions.html"><strong>RNASeq Gene Expressions</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="pathwayanalysis.html"><strong>Pathway Analysis for RNASeq and Microarrays</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="CreateTx2gene.html"><strong>Transcripts to Genes</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="deconvolution.html"><strong>RNA Deconvolution</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="kinship.html"><strong>Kinship and Relatedness</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="metagenomics.html"><strong>Metagenomics Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html"><strong>Installation Guide</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="resources.html"><strong>External Resources</strong></a></li>
</ul>

      <h3>This Page</h3>
      <ul class="this-page-menu">
        <li><a href="_sources/NGS.rst.txt"
               rel="nofollow">Show Source</a></li>
      </ul>
    <div id="searchbox" style="display: none">
        <form class="search" action="search.html" method="get">
                        <div class="search-wrapper">
                        <input type="submit" value="" class="search-left"/>
                        <input class="prettysearch" type="text" name="q" size="18" title="Enter search terms or a module, class or function name."/>
                        <span class="search-right">&nbsp;</span>
                        </div>
          <input type="hidden" name="check_keywords" value="yes" />
          <input type="hidden" name="area" value="default" />
        </form>
    </div>
    <script type="text/javascript">$('#searchbox').show(0);</script>
  </div>
</div>



    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="next-generation-sequencing-toolkit">
<h1><strong>Next Generation Sequencing Toolkit</strong><a class="headerlink" href="#next-generation-sequencing-toolkit" title="Permalink to this headline">¶</a></h1>
<div class="section" id="reads-trimming">
<span id="trimming"></span><h2>Reads Trimming<a class="headerlink" href="#reads-trimming" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="using-trimgalore">
<h2>Using Trimgalore<a class="headerlink" href="#using-trimgalore" title="Permalink to this headline">¶</a></h2>
<p>We need to install <a class="reference internal" href="installation.html#set-trimgalore"><span class="std std-ref">Trimgalore</span></a> and its requirements <a class="reference internal" href="installation.html#set-cutadapt"><span class="std std-ref">Cutadapt</span></a> and <a class="reference internal" href="installation.html#set-fastqc"><span class="std std-ref">FastQC</span></a></p>
<p>Then we remove adaptors and quality trim data as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">trim_galore</span> <span class="o">--</span><span class="n">paired</span> <span class="o">--</span><span class="n">gzip</span> <span class="o">--</span><span class="n">retain_unpaired</span> <span class="o">--</span><span class="n">trim1</span> <span class="o">--</span><span class="n">fastqc</span> <span class="o">--</span><span class="n">fastqc_args</span> <span class="s2">&quot;--outdir fastqc&quot;</span> <span class="o">-</span><span class="n">o</span> <span class="n">galore</span> <span class="o">--</span><span class="n">path_to_cutadapt</span> <span class="n">cutadapt_path</span>  <span class="n">pair_1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">pair_2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div>
<div class="section" id="using-bbduk">
<h2>Using BBDuk<a class="headerlink" href="#using-bbduk" title="Permalink to this headline">¶</a></h2>
<p>A more sensitive trimmer is BBDuk from <a class="reference internal" href="installation.html#set-bbmap"><span class="std std-ref">BBMap</span></a>, but beware trimming is losing data:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bash</span> <span class="n">bbduk</span><span class="o">.</span><span class="n">sh</span> <span class="o">-</span><span class="n">Xmx1g</span> <span class="n">in1</span><span class="o">=</span><span class="n">R1_001</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">in2</span><span class="o">=</span><span class="n">R2_001</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">out</span><span class="o">=</span><span class="n">trimmed_reads</span><span class="o">.</span><span class="n">fq</span> <span class="n">ref</span><span class="o">=</span><span class="n">resources</span><span class="o">/</span><span class="n">adapters</span><span class="o">.</span><span class="n">fa</span> <span class="n">ktrim</span><span class="o">=</span><span class="n">r</span> <span class="n">k</span><span class="o">=</span><span class="mi">23</span> <span class="n">mink</span><span class="o">=</span><span class="mi">11</span> <span class="n">hdist</span><span class="o">=</span><span class="mi">1</span> <span class="n">stats</span><span class="o">=</span><span class="n">stat</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</div>
<div class="section" id="estimating-insert-size">
<h2>Estimating Insert Size<a class="headerlink" href="#estimating-insert-size" title="Permalink to this headline">¶</a></h2>
<p>Here, we estimate the insert size using <a class="reference internal" href="installation.html#set-bbmap"><span class="std std-ref">BBMap</span></a> by aligning 4 million reads to the reference genome, pretty enough for a good estimation.
The output histogram will be written to histmap4m.txt:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bash</span> <span class="n">bbmap</span><span class="o">.</span><span class="n">sh</span> <span class="o">-</span><span class="n">Xmx50g</span>  <span class="n">in1</span><span class="o">=</span><span class="n">pair1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">in2</span><span class="o">=</span><span class="n">pair2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">ihist</span><span class="o">=</span><span class="n">histmap4m</span><span class="o">.</span><span class="n">txt</span> <span class="n">reads</span><span class="o">=</span><span class="mi">4000000</span> <span class="n">ref</span><span class="o">=</span><span class="n">human_genome</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>If you don’t have enough memory, you can use bbmerge, however, bbmap estimation is more accurate as it is based on alignment to reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bash</span> <span class="n">bbmerge</span><span class="o">.</span><span class="n">sh</span> <span class="n">in1</span><span class="o">=</span><span class="n">pair1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">in2</span><span class="o">=</span><span class="n">pair2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">ihist</span><span class="o">=</span><span class="n">hist_merge</span><span class="o">.</span><span class="n">txt</span> <span class="n">reads</span><span class="o">=</span><span class="mi">4000000</span> <span class="n">prefilter</span><span class="o">=</span><span class="mi">2</span> <span class="n">rem</span> <span class="n">extend2</span><span class="o">=</span><span class="mi">100</span>
</pre></div>
</div>
</div>
<div class="section" id="get-unmapped-reads-from-a-bam-file">
<h2>Get Unmapped reads from a BAM file<a class="headerlink" href="#get-unmapped-reads-from-a-bam-file" title="Permalink to this headline">¶</a></h2>
<p>Lets first install <a class="reference internal" href="installation.html#set-samtools"><span class="std std-ref">Samtools</span></a>.</p>
<p>Here, we will get unmapped reads in a BAM file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">unmapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<p>Add -b to get the unmapped reads in a BAM</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">unmapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Or add -c to  count the unmapped reads</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">c</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span>  <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;&gt;</span> <span class="n">unmapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">count</span>
</pre></div>
</div>
</div>
<div class="section" id="get-mapped-reads-from-a-bam-file">
<h2>Get Mapped reads from a BAM file<a class="headerlink" href="#get-mapped-reads-from-a-bam-file" title="Permalink to this headline">¶</a></h2>
<p>Use -F to get the mapped reads as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<p>And similarily, add -b to get the mapped reads in a bam file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>To count mapped reads from a BAM file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">c</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span>  <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">count</span>
</pre></div>
</div>
</div>
</div>


          </div> 
        </div>
      </div>
    <div class="footer">
    <p>
      &copy; Copyright 2018, Sherine Awad.
      Created using <a href="http://www.sphinx-doc.org/">Sphinx</a> 3.2.1 <a href="https://github.com/mga-sphinx/sphinx_adc_theme">ADC Theme</a> 0.1.
    (Revision )
        </p>
    </div>
      <div class="clearer"></div>
    </div>
        <div id="breadcrumbs">
                <strong>Next Generation Sequencing Toolkit</strong>
        </div>
        <script type="text/javascript" charset="utf-8" src="_static/js/toc.js"></script>
  </body>
</html>