<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Rare Variants Association Analysis &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/css/adctheme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <meta name="generator" content="Sphinx/ADC Theme by Mirounga www.mirounga.fr"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="BitByBit  documentation" href="index.html" />
    <link rel="next" title="SNP Array Toolkit" href="snparray.html" />
    <link rel="prev" title="Welcome to BitByBit Bioinformatics Tutorials" href="index.html" />
    <link media="only screen and (max-device-width: 480px)" href="_static/css/mobile.css" type="text/css" rel="stylesheet" /> 
  </head><body>
<div id="docstitle">
    <p>BitByBit  documentation</p>
</div>
<div id="header">
    <div id="title"><h1>Rare Variants Association Analysis</h1></div>
    <ul id="headerButtons">
        <li id="toc_button"><div class="headerButton"><a href="#">Table of Contents</a></div></li>
        <li id="page_buttons">
            <div class="headerButton"><a href="genindex.html" title="General Index" accesskey="I">index</a></div>
            <div class="headerButton"><a href="snparray.html" title="SNP Array Toolkit" accesskey="N">next</a></div>
            <div class="headerButton"><a href="index.html" title="Welcome to BitByBit Bioinformatics Tutorials" accesskey="P">previous</a></div>
        </li>
    </ul>
</div>

<div id="sphinxsidebar">
  <div class="sphinxsidebarwrapper">
        <ul><li class="toctree-l1"><a href="index.html">Main Page</a></li></ul>
        <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Rare Variants Association Analysis</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="snparray.html"><strong>SNP Array Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="wholegenomes.html"><strong>Whole Genomes Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="NGS.html"><strong>Next Generation Sequencing Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseq.html"><strong>RNASeq Tutorial</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html"><strong>Mapping using Salmon</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling1.html"><strong>Samtools Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling2.html"><strong>GATK Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="vcfrandom.html"><strong>VCF Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="SV.html"><strong>Structural Variation</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseqvariants.html"><strong>Variant Calling for RNASeq Data</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="somatics.html"><strong>Somatic Mutations from RNAseq</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="lncRNA.html"><strong>Long nonCoding RNA Prediction</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseqexpressions.html"><strong>RNASeq Gene Expressions</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="pathwayanalysis.html"><strong>Pathway Analysis for RNASeq and Microarrays</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="CreateTx2gene.html"><strong>Transcripts to Genes</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="deconvolution.html"><strong>RNA Deconvolution</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="kinship.html"><strong>Kinship and Relatedness</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="metagenomics.html"><strong>Metagenomics Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html"><strong>Installation Guide</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="resources.html"><strong>External Resources</strong></a></li>
</ul>

      <h3>This Page</h3>
      <ul class="this-page-menu">
        <li><a href="_sources/rvassocanalysis.rst.txt"
               rel="nofollow">Show Source</a></li>
      </ul>
    <div id="searchbox" style="display: none">
        <form class="search" action="search.html" method="get">
                        <div class="search-wrapper">
                        <input type="submit" value="" class="search-left"/>
                        <input class="prettysearch" type="text" name="q" size="18" title="Enter search terms or a module, class or function name."/>
                        <span class="search-right">&nbsp;</span>
                        </div>
          <input type="hidden" name="check_keywords" value="yes" />
          <input type="hidden" name="area" value="default" />
        </form>
    </div>
    <script type="text/javascript">$('#searchbox').show(0);</script>
  </div>
</div>



    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="rare-variants-association-analysis">
<h1><strong>Rare Variants Association Analysis</strong><a class="headerlink" href="#rare-variants-association-analysis" title="Permalink to this headline">¶</a></h1>
<p>Lets install <a class="reference internal" href="installation.html#set-rvtests"><span class="std std-ref">rvtests</span></a> before starting this tutorial.</p>
<p>Instead of testing whether a single variant is associated with a specific phenotype, we can study the cummulative effect of a group of variants in a gene or region on a specific phenotype.
Rare variants association analysis has lots of categories, we will summarize here a few categories:</p>
<div class="section" id="burden-tests">
<h2>Burden Tests<a class="headerlink" href="#burden-tests" title="Permalink to this headline">¶</a></h2>
<p>Burden test is prefered when we have variants in the same direction and the same effect, i.e. variants are risk increasing or risk decreasing. Many methods fall under this category including <strong>CMC</strong>, <strong>CMAT</strong>, etc. Below is an example of using rvtest for burden test using CMC method.</p>
<p>Using rvtest for Burden Test CMC:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
<p>In the above example, we grouped the variants with MAF less than 0.01 per gene,so variants with MAF &lt; 0.01 in each gene are grouped together for the analysis.</p>
<p>If you are interested in specific set of genes instead of all genes, you can specify using –gene parameter as follows</p>
<p>And to specify a specific set of genes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">gene</span> <span class="n">GPR98</span><span class="p">,</span><span class="n">USH2A</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
</div>
<div class="section" id="adaptive-burden-tests">
<h2>Adaptive Burden Tests<a class="headerlink" href="#adaptive-burden-tests" title="Permalink to this headline">¶</a></h2>
<p>Adaptive is more robust than Burden test as it allows for adjusting weights.</p>
<p>KBAC is famous approach that fall under the Adaptive burden test category, and we can run as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">kernel</span> <span class="n">kbac</span>
</pre></div>
</div>
</div>
<div class="section" id="variance-components-tests">
<h2>Variance Components Tests<a class="headerlink" href="#variance-components-tests" title="Permalink to this headline">¶</a></h2>
<p>These tests are powerful if there exist both trait-increasing and trait-decreasing variants or a small fraction of causal variants.</p>
<p>These types of tests are less powerful when the variants are in the same direction.</p>
<p>A famous approach under this category is SKAT, and can be calculated using rvtests as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skat</span>
</pre></div>
</div>
</div>
<div class="section" id="combined-tests">
<h2>Combined Tests<a class="headerlink" href="#combined-tests" title="Permalink to this headline">¶</a></h2>
<p>When we don’t have any prior information about the variants we have, these approaches are robust regardless of the percentage of causal variants and
the existence of both trait-increasing and trait-decreasing variants.</p>
<p>A famous approach under this category is SKATO:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skato</span>
</pre></div>
</div>
</div>
</div>


          </div> 
        </div>
      </div>
    <div class="footer">
    <p>
      &copy; Copyright 2018, Sherine Awad.
      Created using <a href="http://www.sphinx-doc.org/">Sphinx</a> 3.2.1 <a href="https://github.com/mga-sphinx/sphinx_adc_theme">ADC Theme</a> 0.1.
    (Revision )
        </p>
    </div>
      <div class="clearer"></div>
    </div>
        <div id="breadcrumbs">
                <strong>Rare Variants Association Analysis</strong>
        </div>
        <script type="text/javascript" charset="utf-8" src="_static/js/toc.js"></script>
  </body>
</html>