<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Whole Genomes Toolkit &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/css/adctheme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <meta name="generator" content="Sphinx/ADC Theme by Mirounga www.mirounga.fr"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="BitByBit  documentation" href="index.html" />
    <link rel="next" title="Next Generation Sequencing Toolkit" href="NGS.html" />
    <link rel="prev" title="SNP Array Toolkit" href="snparray.html" />
    <link media="only screen and (max-device-width: 480px)" href="_static/css/mobile.css" type="text/css" rel="stylesheet" /> 
  </head><body>
<div id="docstitle">
    <p>BitByBit  documentation</p>
</div>
<div id="header">
    <div id="title"><h1>Whole Genomes Toolkit</h1></div>
    <ul id="headerButtons">
        <li id="toc_button"><div class="headerButton"><a href="#">Table of Contents</a></div></li>
        <li id="page_buttons">
            <div class="headerButton"><a href="genindex.html" title="General Index" accesskey="I">index</a></div>
            <div class="headerButton"><a href="NGS.html" title="Next Generation Sequencing Toolkit" accesskey="N">next</a></div>
            <div class="headerButton"><a href="snparray.html" title="SNP Array Toolkit" accesskey="P">previous</a></div>
        </li>
    </ul>
</div>

<div id="sphinxsidebar">
  <div class="sphinxsidebarwrapper">
        <ul><li class="toctree-l1"><a href="index.html">Main Page</a></li></ul>
        <p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="rvassocanalysis.html"><strong>Rare Variants Association Analysis</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="snparray.html"><strong>SNP Array Toolkit</strong></a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Whole Genomes Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="NGS.html"><strong>Next Generation Sequencing Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseq.html"><strong>RNASeq Tutorial</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html"><strong>Mapping using Salmon</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling1.html"><strong>Samtools Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="variantcalling2.html"><strong>GATK Variant Calling</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="vcfrandom.html"><strong>VCF Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="SV.html"><strong>Structural Variation</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="RNAseqvariants.html"><strong>Variant Calling for RNASeq Data</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="somatics.html"><strong>Somatic Mutations from RNAseq</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="lncRNA.html"><strong>Long nonCoding RNA Prediction</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseqexpressions.html"><strong>RNASeq Gene Expressions</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="pathwayanalysis.html"><strong>Pathway Analysis for RNASeq and Microarrays</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="CreateTx2gene.html"><strong>Transcripts to Genes</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="deconvolution.html"><strong>RNA Deconvolution</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="kinship.html"><strong>Kinship and Relatedness</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="metagenomics.html"><strong>Metagenomics Toolkit</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="installation.html"><strong>Installation Guide</strong></a></li>
<li class="toctree-l1"><a class="reference internal" href="resources.html"><strong>External Resources</strong></a></li>
</ul>

      <h3>This Page</h3>
      <ul class="this-page-menu">
        <li><a href="_sources/wholegenomes.rst.txt"
               rel="nofollow">Show Source</a></li>
      </ul>
    <div id="searchbox" style="display: none">
        <form class="search" action="search.html" method="get">
                        <div class="search-wrapper">
                        <input type="submit" value="" class="search-left"/>
                        <input class="prettysearch" type="text" name="q" size="18" title="Enter search terms or a module, class or function name."/>
                        <span class="search-right">&nbsp;</span>
                        </div>
          <input type="hidden" name="check_keywords" value="yes" />
          <input type="hidden" name="area" value="default" />
        </form>
    </div>
    <script type="text/javascript">$('#searchbox').show(0);</script>
  </div>
</div>



    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="whole-genomes-toolkit">
<h1><strong>Whole Genomes Toolkit</strong><a class="headerlink" href="#whole-genomes-toolkit" title="Permalink to this headline">¶</a></h1>
<div class="section" id="in-progress">
<h2>In Progress<a class="headerlink" href="#in-progress" title="Permalink to this headline">¶</a></h2>
<p>Here, we will show various operations in whole genome Complete Genomics or BGI.</p>
<div class="section" id="evidence-to-sam">
<h3>Evidence to Sam<a class="headerlink" href="#evidence-to-sam" title="Permalink to this headline">¶</a></h3>
<p>Convert to bam from cgatools, and then we can view the sorted bam in IGV:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cgatools</span> <span class="n">evidence2sam</span> \
<span class="o">--</span><span class="n">evidence</span><span class="o">-</span><span class="n">dnbs</span><span class="o">=</span><span class="n">evidenceDnbs</span><span class="o">-</span><span class="n">chr9</span><span class="o">-</span><span class="n">GS000004829</span><span class="o">-</span><span class="n">ASM</span><span class="o">.</span><span class="n">tsv</span><span class="o">.</span><span class="n">bz2</span> \
<span class="o">--</span><span class="n">beta</span> \
<span class="o">--</span><span class="n">reference</span><span class="o">=</span><span class="n">build37</span><span class="o">.</span><span class="n">crr</span> <span class="o">|</span> <span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">uS</span> <span class="o">-</span> <span class="o">|</span> \
<span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">T</span> <span class="n">gs4829_9</span><span class="o">.</span><span class="n">sorted</span> <span class="o">-</span><span class="n">o</span> <span class="n">chr9</span><span class="o">-</span><span class="n">GS000004829</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">&amp;&amp;</span> <span class="n">samtools</span> <span class="n">index</span> <span class="n">chr9</span><span class="o">-</span><span class="n">GS000004829</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Here we used the evidence file from cgatools output, and used evidence2sam to convert it to sam then sorted bam.</p>
</div>
<div class="section" id="cgatools-tsv-to-annovar-format">
<h3>CGAtools tsv to annovar format<a class="headerlink" href="#cgatools-tsv-to-annovar-format" title="Permalink to this headline">¶</a></h3>
<p>Convert from cgatools .tsv to Annovar format to further annotate with Annovar:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">convert2annovar</span><span class="o">.</span><span class="n">pl</span> <span class="o">-</span><span class="nb">format</span> <span class="n">cg</span> <span class="o">-</span><span class="n">out</span> <span class="n">patientX</span><span class="o">.</span><span class="n">annovar</span><span class="o">.</span><span class="n">input</span>  <span class="n">var</span><span class="o">-</span><span class="n">GSPatientX</span><span class="o">-</span><span class="n">ASM</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
<p>Now, we can use patientX.annovar.input in Annovar:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">table_annovar</span><span class="o">.</span><span class="n">pl</span> <span class="n">patientX</span><span class="o">.</span><span class="n">annovar</span><span class="o">.</span><span class="n">input</span>  <span class="n">humandb</span><span class="o">/</span> <span class="o">-</span><span class="n">buildver</span> <span class="n">hg19</span> <span class="o">-</span><span class="n">out</span> <span class="n">patientX</span> <span class="o">-</span><span class="n">remove</span> <span class="o">-</span><span class="n">protocol</span> <span class="n">refGene</span><span class="p">,</span><span class="n">cytoBand</span><span class="p">,</span><span class="n">gnomad_exome</span><span class="p">,</span><span class="n">exac03</span><span class="p">,</span><span class="n">avsnp147</span><span class="p">,</span>\
   <span class="n">dbnsfp30a</span><span class="p">,</span><span class="n">clinvar_20170130</span><span class="p">,</span><span class="n">mitimpact2</span><span class="p">,</span><span class="n">revel</span> <span class="o">-</span><span class="n">operation</span> <span class="n">g</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span>  <span class="o">-</span><span class="n">nastring</span> <span class="o">.</span> <span class="o">-</span><span class="n">csvout</span> <span class="o">-</span><span class="n">polish</span> <span class="o">-</span><span class="n">xref</span> <span class="n">example</span><span class="o">/</span><span class="n">gene_xref</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>where example/gene_xref.txt is the xref file found in example folder of Annovar.</p>
</div>
</div>
</div>


          </div> 
        </div>
      </div>
    <div class="footer">
    <p>
      &copy; Copyright 2018, Sherine Awad.
      Created using <a href="http://www.sphinx-doc.org/">Sphinx</a> 3.2.1 <a href="https://github.com/mga-sphinx/sphinx_adc_theme">ADC Theme</a> 0.1.
    (Revision )
        </p>
    </div>
      <div class="clearer"></div>
    </div>
        <div id="breadcrumbs">
                <strong>Whole Genomes Toolkit</strong>
        </div>
        <script type="text/javascript" charset="utf-8" src="_static/js/toc.js"></script>
  </body>
</html>