

<!doctype html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Next Generation Sequencing Toolkit &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/bizstyle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="VCF Toolkit" href="vcfrandom.html" />
    <link rel="prev" title="Whole Genomes Toolkit" href="wholegenomes.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <!--[if lt IE 9]>
    <script type="text/javascript" src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="vcfrandom.html" title="VCF Toolkit"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Toolkit"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#"><strong>Next Generation Sequencing Toolkit</strong></a><ul>
<li><a class="reference internal" href="#reads-trimming">Reads Trimming</a></li>
<li><a class="reference internal" href="#estimating-insert-size">Estimating Insert Size</a></li>
<li><a class="reference internal" href="#get-mapped-reads-from-a-bam-file">Get Mapped reads from a BAM file</a></li>
<li><a class="reference internal" href="#id1">Get Mapped reads from a BAM file</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="wholegenomes.html"
                        title="previous chapter"><strong>Whole Genomes Toolkit</strong></a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="vcfrandom.html"
                        title="next chapter"><strong>VCF Toolkit</strong></a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/NGS.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="next-generation-sequencing-toolkit">
<h1><strong>Next Generation Sequencing Toolkit</strong><a class="headerlink" href="#next-generation-sequencing-toolkit" title="Permalink to this headline">¶</a></h1>
<div class="section" id="reads-trimming">
<h2>Reads Trimming<a class="headerlink" href="#reads-trimming" title="Permalink to this headline">¶</a></h2>
<p>We need to install Trimgalore <a class="reference internal" href="installation.html#set-trimgalore"><span class="std std-ref">Install Trimgalore</span></a> and its requirements Cutadapt  <a class="reference internal" href="installation.html#set-cutadapt"><span class="std std-ref">Install Cutadapt:</span></a> and FastQC <a class="reference internal" href="installation.html#set-fastqc"><span class="std std-ref">Install FastQC:</span></a></p>
<p>Then we remove adaptors and quality trim data as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">trim_galore</span> <span class="o">--</span><span class="n">paired</span> <span class="o">--</span><span class="n">gzip</span> <span class="o">--</span><span class="n">retain_unpaired</span> <span class="o">--</span><span class="n">trim1</span> <span class="o">--</span><span class="n">fastqc</span> <span class="o">--</span><span class="n">fastqc_args</span> <span class="s2">&quot;--outdir fastqc&quot;</span> <span class="o">-</span><span class="n">o</span> <span class="n">galore</span> <span class="o">--</span><span class="n">path_to_cutadapt</span> <span class="n">cutadapt_path</span>  <span class="n">pair_1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">pair_2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div>
<div class="section" id="estimating-insert-size">
<h2>Estimating Insert Size<a class="headerlink" href="#estimating-insert-size" title="Permalink to this headline">¶</a></h2>
<p>Here, we estimate the insert size using bbmap by aligning 4 million reads to the reference genome, pretty enough for a good estimation. The output histogram will be written to histmap1mtxt:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bash</span> <span class="n">bbmap</span><span class="o">.</span><span class="n">sh</span> <span class="o">-</span><span class="n">Xmx50g</span>  <span class="n">in1</span><span class="o">=</span><span class="n">pair1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">in2</span><span class="o">=</span><span class="n">pair2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">ihist</span><span class="o">=</span><span class="n">histmap4m</span><span class="o">.</span><span class="n">txt</span> <span class="n">reads</span><span class="o">=</span><span class="mi">4000000</span> <span class="n">ref</span><span class="o">=</span><span class="n">human_genome</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>If you don’t have enough memory, you can use bbmerge, however, bbmap estimation is more accurate as it is based on alignment to reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bash</span> <span class="n">bbmerge</span><span class="o">.</span><span class="n">sh</span> <span class="n">in1</span><span class="o">=</span> <span class="n">in1</span><span class="o">=</span><span class="n">pair1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">in2</span><span class="o">=</span><span class="n">pair2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">ihist</span><span class="o">=</span><span class="n">hist_merge</span><span class="o">.</span><span class="n">txt</span> <span class="n">reads</span><span class="o">=</span><span class="mi">4000000</span> <span class="n">prefilter</span><span class="o">=</span><span class="mi">2</span> <span class="n">rem</span> <span class="n">extend2</span><span class="o">=</span><span class="mi">100</span>
</pre></div>
</div>
</div>
<div class="section" id="get-mapped-reads-from-a-bam-file">
<h2>Get Mapped reads from a BAM file<a class="headerlink" href="#get-mapped-reads-from-a-bam-file" title="Permalink to this headline">¶</a></h2>
<p>Here, we will get mapped reads in a BAM file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<p>Add -b to get the read in a BAM</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Or add -c to  count the mapped reads</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">c</span> <span class="o">-</span><span class="n">f</span> <span class="mi">4</span>  <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;&gt;</span>  <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">count</span>
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h2>Get Mapped reads from a BAM file<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>Use -F to get the mapped reads as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<p>And similarily, add -b to get the mapped reads in a bam file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span> <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;</span> <span class="n">mapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>To count unmapped reads from a BAM file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="n">c</span> <span class="o">-</span><span class="n">F</span> <span class="mi">4</span>  <span class="n">alignments</span><span class="o">.</span><span class="n">bam</span> <span class="o">&gt;&gt;</span>  <span class="n">unmapped</span><span class="o">.</span><span class="n">reads</span><span class="o">.</span><span class="n">count</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="vcfrandom.html" title="VCF Toolkit"
             >next</a> |</li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Toolkit"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Sherine Awad.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.0b1.
    </div>
  </body>
</html>