

<!doctype html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>RNASeq Gene Expressions &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/bizstyle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Pathway Analysis for RNASeq and Microarrays" href="pathwayanalysis.html" />
    <link rel="prev" title="Long nonCoding RNA Prediction" href="lncRNA.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <!--[if lt IE 9]>
    <script type="text/javascript" src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="pathwayanalysis.html" title="Pathway Analysis for RNASeq and Microarrays"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="lncRNA.html" title="Long nonCoding RNA Prediction"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#"><strong>RNASeq Gene Expressions</strong></a><ul>
<li><a class="reference internal" href="#in-progress">In Progress</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="lncRNA.html"
                        title="previous chapter"><strong>Long nonCoding RNA Prediction</strong></a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="pathwayanalysis.html"
                        title="next chapter"><strong>Pathway Analysis for RNASeq and Microarrays</strong></a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/rnaseqexpressions.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="rnaseq-gene-expressions">
<h1><strong>RNASeq Gene Expressions</strong><a class="headerlink" href="#rnaseq-gene-expressions" title="Permalink to this headline">¶</a></h1>
<p>See Salmon quantifications section</p>
<div class="section" id="in-progress">
<h2>In Progress<a class="headerlink" href="#in-progress" title="Permalink to this headline">¶</a></h2>
<p>We need to write R script and use EgdeR.
Lets first declare our libraries</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">library</span><span class="p">(</span><span class="n">tximport</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">rjson</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;edgeR&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;AnnotationDbi&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;org.Hs.eg.db&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">magrittr</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">pathview</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">gage</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">gageData</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">table</span><span class="p">)</span>
</pre></div>
</div>
<p>Now, lets read salmon quantifications files</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>samples &lt;- read.table(file.path(&quot;/home/sherine/work/rnaseq/kaz&quot;, &quot;samples.txt&quot;), header = FALSE)
files &lt;- file.path(&quot;/home/sherine/work/rnaseq&quot;, &quot;kaz&quot;, samples$V1, &quot;quant.sf&quot;)
print (&#39;Files are :&#39;)
print (files)
names(files) &lt;- paste0(&quot;sample&quot;, 1:20)
</pre></div>
</div>
<p>Lets convert salmon transcripts to gene names using NCBItx2gene.csv. See <a class="reference internal" href="CreateTx2gene.html#txtogene"><span class="std std-ref">How to convert from transcripts to Genes ids</span></a> for how to generate this file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>tx2gene &lt;- read.csv(file.path(&quot;/home/sherine/work/rnaseq/kaz&quot;, &quot;NCBItx2gene.csv&quot;))
txi &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene,ignoreTxVersion = TRUE, lengthCol=length)
cts &lt;- txi$counts
write.csv(cts, &quot;counts.csv&quot;)
</pre></div>
</div>
<p>We define our groups and create our DGEList</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">group</span> <span class="o">=</span> <span class="n">factor</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="n">rep</span><span class="p">(</span><span class="s2">&quot;Control&quot;</span><span class="p">,</span> <span class="mi">10</span><span class="p">),</span> <span class="n">rep</span><span class="p">(</span><span class="s2">&quot;Tumor&quot;</span><span class="p">,</span><span class="mi">10</span><span class="p">))</span> <span class="p">)</span>
<span class="n">dge</span> <span class="o">=</span> <span class="n">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="o">=</span><span class="n">cts</span><span class="p">,</span> <span class="n">genes</span><span class="o">=</span> <span class="n">rownames</span><span class="p">(</span><span class="n">data</span><span class="p">),</span> <span class="n">group</span><span class="o">=</span><span class="n">group</span><span class="p">)</span>
<span class="n">countsPerMillion</span> <span class="o">&lt;-</span> <span class="n">cpm</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<span class="n">summary</span><span class="p">(</span><span class="n">countsPerMillion</span><span class="p">)</span>
<span class="n">countCheck</span> <span class="o">&lt;-</span> <span class="n">countsPerMillion</span> <span class="o">&gt;</span> <span class="mi">1</span>
<span class="n">summary</span><span class="p">(</span><span class="n">countCheck</span><span class="p">)</span>
</pre></div>
</div>
<p>Filtering</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">keep</span> <span class="o">&lt;-</span> <span class="n">which</span><span class="p">(</span><span class="n">rowSums</span><span class="p">(</span><span class="n">countCheck</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="mi">2</span><span class="p">)</span>
<span class="n">dge</span> <span class="o">&lt;-</span> <span class="n">dge</span><span class="p">[</span><span class="n">keep</span><span class="p">,]</span>
<span class="n">summary</span><span class="p">(</span><span class="n">cpm</span><span class="p">(</span><span class="n">dge</span><span class="p">))</span>
</pre></div>
</div>
<p>Calculate expressions, see this interesting post about the CPM, RPKM , and FPKM differences <a class="reference external" href="https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/">https://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dge</span> <span class="o">&lt;-</span> <span class="n">calcNormFactors</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">&quot;TMM&quot;</span><span class="p">)</span>
<span class="n">summary</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<span class="n">dge</span> <span class="o">&lt;-</span> <span class="n">estimateCommonDisp</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<span class="n">dge</span> <span class="o">&lt;-</span> <span class="n">estimateTagwiseDisp</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<span class="n">dge</span> <span class="o">&lt;-</span> <span class="n">estimateTrendedDisp</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<span class="n">et</span> <span class="o">&lt;-</span> <span class="n">exactTest</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">pair</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">&quot;Control&quot;</span><span class="p">,</span> <span class="s2">&quot;Tumor&quot;</span><span class="p">))</span>
</pre></div>
</div>
<p>Note that here we can filter using pvalues, but if you plan to use Gage for pathways, Gage takes all the data with no filter as a background otherwise you will get lots of nulls:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">etp</span> <span class="o">&lt;-</span> <span class="n">topTags</span><span class="p">(</span><span class="n">et</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span> <span class="mi">100000</span><span class="p">,</span> <span class="n">adjust</span><span class="o">.</span><span class="n">method</span><span class="o">=</span><span class="s2">&quot;BH&quot;</span><span class="p">,</span> <span class="n">sort</span><span class="o">.</span><span class="n">by</span><span class="o">=</span><span class="s2">&quot;PValue&quot;</span><span class="p">,</span> <span class="n">p</span><span class="o">.</span><span class="n">value</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
<p>If you plan to use xcells for RNAseq deconvolution, xcell requires RPKM instead of CPM, so use</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>length = txi$length
dge = DGEList(counts=cts, genes= rownames(data), group=group)
dge &lt;- calcNormFactors(dge, method=&quot;TMM&quot;)
summary(dge)
rpkm = rpkm(dge, length)
write.csv(rpkm, &quot;rpkm.csv&quot;)
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="pathwayanalysis.html" title="Pathway Analysis for RNASeq and Microarrays"
             >next</a> |</li>
        <li class="right" >
          <a href="lncRNA.html" title="Long nonCoding RNA Prediction"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Sherine Awad.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.0b1.
    </div>
  </body>
</html>