
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Rare Variants Association Analysis &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Whole Genomes Analysis" href="wholegenomes.html" />
    <link rel="prev" title="Welcome to BitByBit’s documentation!" href="index.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Analysis"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to BitByBit’s documentation!"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Welcome to BitByBit’s documentation!</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="wholegenomes.html"
                        title="next chapter">Whole Genomes Analysis</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/rvassocanalysis.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="rare-variants-association-analysis">
<h1><strong>Rare Variants Association Analysis</strong><a class="headerlink" href="#rare-variants-association-analysis" title="Permalink to this headline">¶</a></h1>
<p>Instead of testing whether a single variant is associated with a specific phenotype, we can study the cummulative effect of a group of variants in a gene or region on a specific phenotype.
Rare variants association analysis has lots of categories, we will summarize here a few categories:</p>
<ol class="arabic simple">
<li>Burden Test</li>
</ol>
<p>Burden test is prefered when we have variants in the same direction and the same effect, i.e. variants are risk increasing or risk decreasing. Many methods fall under this category including <strong>CMC</strong>, <strong>CMAT</strong>, etc. Below is an example of using rvtest for burden test using CMC method.</p>
<p>Using rvtest for Burden Test CMC:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
<p>In the above example, we grouped the variants with MAF less than 0.01 per gene,so variants with MAF &lt; 0.01 in each gene are grouped together for the analysis.</p>
<p>If you are interested in specific set of genes instead of all genes, you can specify using –gene parameter as follows</p>
<p>And to specify a specific set of genes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">gene</span> <span class="n">GPR98</span><span class="p">,</span><span class="n">USH2A</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li>Adaptive Burden Test</li>
</ol>
<p>Adaptive is more robust than Burden test as it allows for adjusting weights.</p>
<p>KBAC is famous approach that fall under the Adaptive burden test category, and we can run as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">kernel</span> <span class="n">kbac</span>
</pre></div>
</div>
<ol class="arabic simple" start="3">
<li>Variance Components Tests</li>
</ol>
<p>These tests are powerful if there exist both trait-increasing and trait-decreasing variants or a small fraction of causal variants.</p>
<p>These types of tests are less powerful when the variants are in the same direction.</p>
<p>A famous approach under this category is SKAT, and can be calculated using rvtests as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skat</span>
</pre></div>
</div>
<ol class="arabic simple" start="4">
<li>Combined Tests</li>
</ol>
<dl class="docutils">
<dt>When we don’t have any prior information about the variants we have, these approaches are robust regardless of the percentage of causal variants and</dt>
<dd>the existence of both trait-increasing and trait-decreasing variants.</dd>
</dl>
<p>A famous approach under this category is SKATO:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skato</span>
</pre></div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Analysis"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to BitByBit’s documentation!"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Sherine Awad.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.0b1.
    </div>
  </body>
</html>