
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>Rare Variants Association Analysis &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Whole Genomes Toolkit" href="wholegenomes.html" />
    <link rel="prev" title="Welcome to BitByBit’s documentation!" href="index.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Toolkit"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to BitByBit’s documentation!"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#"><strong>Rare Variants Association Analysis</strong></a><ul>
<li><a class="reference internal" href="#burden-tests">Burden Tests</a></li>
<li><a class="reference internal" href="#adaptive-burden-tests">Adaptive Burden Tests</a></li>
<li><a class="reference internal" href="#variance-components-tests">Variance Components Tests</a></li>
<li><a class="reference internal" href="#combined-tests">Combined Tests</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Welcome to BitByBit’s documentation!</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="wholegenomes.html"
                        title="next chapter"><strong>Whole Genomes Toolkit</strong></a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/rvassocanalysis.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="index.html">Docs</a></li>
              
              <li><strong>Rare Variants Association Analysis</strong></li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="rare-variants-association-analysis">
<h1><strong>Rare Variants Association Analysis</strong><a class="headerlink" href="#rare-variants-association-analysis" title="Permalink to this headline">¶</a></h1>
<p>Lets <a class="reference internal" href="installation.html#set-rvtests"><span class="std std-ref">Install rvtests</span></a> before starting this tutorial.</p>
<p>Instead of testing whether a single variant is associated with a specific phenotype, we can study the cummulative effect of a group of variants in a gene or region on a specific phenotype.
Rare variants association analysis has lots of categories, we will summarize here a few categories:</p>
<div class="section" id="burden-tests">
<h2>Burden Tests<a class="headerlink" href="#burden-tests" title="Permalink to this headline">¶</a></h2>
<p>Burden test is prefered when we have variants in the same direction and the same effect, i.e. variants are risk increasing or risk decreasing. Many methods fall under this category including <strong>CMC</strong>, <strong>CMAT</strong>, etc. Below is an example of using rvtest for burden test using CMC method.</p>
<p>Using rvtest for Burden Test CMC:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
<p>In the above example, we grouped the variants with MAF less than 0.01 per gene,so variants with MAF &lt; 0.01 in each gene are grouped together for the analysis.</p>
<p>If you are interested in specific set of genes instead of all genes, you can specify using –gene parameter as follows</p>
<p>And to specify a specific set of genes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">gene</span> <span class="n">GPR98</span><span class="p">,</span><span class="n">USH2A</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span>
</pre></div>
</div>
</div>
<div class="section" id="adaptive-burden-tests">
<h2>Adaptive Burden Tests<a class="headerlink" href="#adaptive-burden-tests" title="Permalink to this headline">¶</a></h2>
<p>Adaptive is more robust than Burden test as it allows for adjusting weights.</p>
<p>KBAC is famous approach that fall under the Adaptive burden test category, and we can run as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">kernel</span> <span class="n">kbac</span>
</pre></div>
</div>
</div>
<div class="section" id="variance-components-tests">
<h2>Variance Components Tests<a class="headerlink" href="#variance-components-tests" title="Permalink to this headline">¶</a></h2>
<p>These tests are powerful if there exist both trait-increasing and trait-decreasing variants or a small fraction of causal variants.</p>
<p>These types of tests are less powerful when the variants are in the same direction.</p>
<p>A famous approach under this category is SKAT, and can be calculated using rvtests as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skat</span>
</pre></div>
</div>
</div>
<div class="section" id="combined-tests">
<h2>Combined Tests<a class="headerlink" href="#combined-tests" title="Permalink to this headline">¶</a></h2>
<p>When we don’t have any prior information about the variants we have, these approaches are robust regardless of the percentage of causal variants and
the existence of both trait-increasing and trait-decreasing variants.</p>
<p>A famous approach under this category is SKATO:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">rvtest</span> <span class="o">--</span><span class="n">inVcf</span> <span class="n">samples</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">pheno</span> <span class="n">samples</span><span class="o">.</span><span class="n">ped</span> <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span> <span class="o">--</span><span class="n">out</span> <span class="n">cmc</span> <span class="o">--</span><span class="n">geneFile</span> <span class="n">refFlat_hg19</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">skato</span>
</pre></div>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="index.html" title="previous chapter (use the left arrow)">Welcome to BitByBit’s documentation!</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="wholegenomes.html" title="next chapter (use the right arrow)"><strong>Whole Genomes Toolkit</strong></a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="wholegenomes.html" title="Whole Genomes Toolkit"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Welcome to BitByBit’s documentation!"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2018, Sherine Awad. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>