Search.setIndex({docnames:["CreateTx2gene","NGS","RNAseqvariants","index","lncRNA","pathwayanalysis","rvassocanalysis","somatics","vcfrandom","wholegenomes"],envversion:50,filenames:["CreateTx2gene.rst","NGS.rst","RNAseqvariants.rst","index.rst","lncRNA.rst","pathwayanalysis.rst","rvassocanalysis.rst","somatics.rst","vcfrandom.rst","wholegenomes.rst"],objects:{},objnames:{},objtypes:{},terms:{"final":5,"function":5,"long":3,"new":[5,8],"null":7,"true":[2,5],"var":9,And:[2,5,6],The:1,Then:[2,5,7],These:6,Use:0,Using:[2,6],about:6,abov:6,accur:1,adapt:6,addorreplacereadgroup:2,adjust:6,after:2,align:[1,2],all:[6,8],allow:6,allow_n_cigar_read:2,allsampl:8,allsamples_chr1:8,analysi:[2,3],analyzecovari:2,ani:6,annot:[0,9],annovar:9,appli:5,approach:6,archiv:4,arg_fil:7,around:2,asm:9,associ:3,assum:2,author:3,avsnp147:9,awad:3,bam:[2,9],base:1,baserecalibr:2,bbmap:1,bbmerg:1,bcftool:8,befor:2,below:6,beta:9,better:2,bgi:9,bgzip:8,both:6,bqsr:2,build37:9,buildver:9,bunch:8,burden:6,bz2:9,calcul:6,call:3,caller:[2,8],can:[1,6,9],care:2,categori:6,causal:6,cgatool:9,charact:5,check:2,chr1:8,chr9:9,chromosom:8,cigar:2,clinvar_20170130:9,cmat:6,cmc:6,column:0,com:4,combin:6,combinevari:7,complet:9,compon:6,configurationfil:[2,7],contain:[4,7],content:3,convers:5,convert2annovar:9,convert:9,coordin:2,creat:[0,2,7],create_index:2,crr:9,csv:[0,5],csvout:9,cummul:6,current:2,cytoband:9,dat:4,data:[3,5,7],dbnsfp30a:9,dbsnp138:2,decreas:6,dedup:2,defin:5,detect:[2,7],differ:8,dir:5,direct:6,dlog4j:[2,7],dnb:9,don:[1,6],done:8,dontusesoftclippedbas:2,download:0,downregul:5,duplic:2,e00461_116:2,e00461_116_gw170602261_1:2,each:6,effect:6,enough:1,entrez:5,entrezid:5,etc:6,evid:9,evidence2sam:9,evidencednb:9,exac03:9,exampl:[6,9],exclud:8,exist:6,extend2:1,fall:6,fals:5,famou:6,fasta:[2,4,7],fastq:[1,2],few:6,file1:8,file2:8,file3:8,file:[0,2,7,8,9],filter:[5,8],filteredareuncal:7,filteredrecordsmergetyp:7,first:[2,7],foldchang:5,folder:9,follow:[2,6,7],format:[4,9],found:9,fraction:6,frame:5,frequpp:6,from:[2,3,8,9],ftp:0,further:[2,9],gage:5,gatk:[2,7],gene:[4,5,6],gene_xref:9,genefil:6,geneid:0,gener:[2,3],genom:[1,3],genomeanalysistk:[2,7],genomedir:2,genomefastafil:2,genomegener:2,get:5,gff:0,github:4,gnomad_exom:9,gobpr:5,gobpset:5,good:1,gov:0,gpr98:6,grch37:0,grch37_latest:0,grch37_latest_genom:0,greater:5,group:[6,8],gs000004829:9,gs4829_9:9,gset:5,gspatientx:9,gtf:4,gzvcf:8,h_sapien:0,haplotyp:2,haplotypecal:2,has:6,have:[1,2,6],head:[0,5],here:[0,1,2,6,7,8,9],hg19:9,hg38:[2,7],hg38index:2,hist_merg:1,histmap1mtxt:1,histmap4m:1,histogram:1,homo_sapiens_assembly38:2,howev:1,hsa:5,http:4,human_genom:1,human_hexam:4,human_train:4,humandb:9,idtyp:5,idx:5,igv:9,ihist:1,illumina:2,in1:1,in2:1,includ:6,increas:6,indel:2,index:[2,3,9],inform:6,input:9,instal:4,instead:6,interest:[2,6],interv:2,invcf:6,jar:[2,7],java:[2,7],kbac:6,keep_if_any_unfilt:7,kegg:5,keggr:5,keggresidsdn:5,keggresidsup:5,keggrespathwaysdn:5,keggrespathwaysup:5,kei:0,kernel:6,keytyp:0,known:2,known_indel:2,knownsit:2,lappli:5,later:5,less:[5,6],let:[2,5],list:7,lncscore:4,log4j2:[2,7],logfc:5,lot:6,maf:6,makesitesonlyvcf:7,maketxdbfromgff:0,mani:6,mark:2,markdupl:2,megr:8,memori:1,merg:8,method:6,metric:2,million:1,mills_and_1000g_gold_standard:2,minn:7,mitimpact2:9,modul:3,more:[1,6],mutat:[2,3],mutect2:2,mytranscript:4,myvcf:8,name:[5,7],namefold:5,nastr:9,ncbi:0,necessari:2,need:[2,5,7,8],next:3,nih:0,nlm:0,noncod:3,normal1:7,normal2:7,normal3:7,normal:7,now:[2,9],one:8,onli:7,oper:[8,9],our:[2,4],ourlncrna:4,out:[2,6,8,9],output:[1,2,9],page:3,pair1:[1,2],pair2:[1,2],panel:7,paramet:6,pass:[2,8],pathview:5,pathwai:5,patientx:9,pdf:2,ped:6,per:[6,8],perced:7,percentag:6,pheno:6,phenotyp:6,picard:[2,7],pid:5,pipelin:[2,7],plot:[2,5],plot_pathwai:5,polish:9,pon:7,pon_siteonli:7,post:2,post_recal_data:2,power:6,pre:2,predict:3,prefer:6,prefilt:1,pretti:1,printread:2,prior:6,protocol:9,python:4,queri:8,rare:3,read:[1,2],readfilesin:2,realign:2,realignertargetcr:2,reassignonemappingqu:2,recablir:2,recal_data:2,recalibration_plot:2,recod:8,ref:[1,8],refer:[1,2,9],refflat_hg19:6,refgen:9,refseq:0,refseq_identifi:0,regardless:6,region:6,regulari:8,rem:1,remov:[8,9],res:5,result:4,revel:9,rg_added_sort:2,rgid:2,rglb:2,rgpl:2,rgpu:2,rgsm:2,risk:6,rmqf:2,rmqt:2,rna:3,rnaseq:3,robust:6,row_numb:5,rownam:5,run:6,runmod:2,runthreadn:2,rvtest:6,s1_1:2,s1_2:2,sam:[2,9],same:[5,6],sampl:[2,6,8],samtool:9,sappli:5,search:3,select:0,sequenc:3,set:[5,6],setkei:7,sherin:3,show:9,sigmet:5,signatur:5,silent:2,singl:6,site:7,sjdbfilechrstartend:2,sjdboverhang:2,skat:6,skato:6,small:6,somat:[2,3],some:[2,5],sort:[8,9],sp1483:2,sp1483_s13:2,speci:5,specif:6,specifi:6,split:[2,8],splitncigarread:2,stand_call_conf:2,star:2,start:5,step:2,stop:5,studi:6,sub:5,subset:8,substr:5,summar:6,tab:2,tabix:8,tabl:2,table_annovar:9,tar:4,tbl_df:5,test:6,than:[2,6],thi:[2,6],thread:2,tmpdn:5,tmpup:5,togeth:6,toolkit:3,top:5,trait:6,transcript:4,tsv:[4,9],tutori:2,tx2gene:0,txdb:0,txname:0,txt:[1,6,9],type:6,under:6,upregul:5,use:[1,2,4,8,9],used:[2,9],ush2a:6,using:[1,6],validation_string:2,varianc:6,variant:[3,8],variou:9,vcf:[2,3,6,7],vcftool:8,view:9,weight:6,wget:[0,4],wglab:4,when:6,where:[2,9],whether:[2,6],which:[2,4],whole:3,write:[0,5],written:1,xml:[2,7],xmx50g:1,xref:9,xzvf:4,you:[1,2,6],your:0},titles:["&lt;no title&gt;","Next Generation Sequencing Toolkit","Variant Calling for RNASeq Data","Welcome to BitByBit&#8217;s documentation!","Long nonCoding RNA Prediction","&lt;no title&gt;","<strong>Rare Variants Association Analysis</strong>","Somatic Mutations from RNAseq","VCF Toolkit","Whole Genomes Toolkit"],titleterms:{"long":4,analysi:6,associ:6,bitbybit:3,call:2,data:2,document:3,estim:1,from:7,gener:1,genom:9,indic:3,insert:1,mutat:7,next:1,noncod:4,predict:4,rare:6,rna:4,rnaseq:[2,7],sequenc:1,size:1,somat:7,tabl:3,toolkit:[1,8,9],variant:[2,6],vcf:8,welcom:3,whole:9}})