Search.setIndex({docnames:["CreateTx2gene","NGS","RNAseqvariants","index","installation","lncRNA","pathwayanalysis","rvassocanalysis","somatics","vcfrandom","wholegenomes"],envversion:53,filenames:["CreateTx2gene.rst","NGS.rst","RNAseqvariants.rst","index.rst","installation.rst","lncRNA.rst","pathwayanalysis.rst","rvassocanalysis.rst","somatics.rst","vcfrandom.rst","wholegenomes.rst"],objects:{},objnames:{},objtypes:{},terms:{"final":6,"function":6,"long":3,"new":[6,9],"null":8,"true":[2,6],"var":10,And:[2,6,7],The:1,Then:[1,2,6,8],These:7,Use:0,Using:[2,7],about:7,abov:7,accur:1,adaptor:1,addorreplacereadgroup:2,adjust:7,after:2,align:[1,2],all:[7,9],allow:7,allow_n_cigar_read:2,allsampl:9,allsamples_chr1:9,analysi:3,analyz:2,analyzecovari:2,ani:7,annot:10,annovar:[],appli:6,approach:7,archiv:4,arg_fil:8,around:2,articl:4,asm:10,assist:3,associ:3,assum:2,author:3,avsnp147:10,awad:3,babraham:4,bam:[2,10],base:1,baserecalibr:2,bash:1,bbmap:1,bbmerg:1,bcftool:9,befor:[2,7],below:7,beta:10,better:2,bgi:10,bgzip:9,bioconda:4,bioinformat:4,both:7,bqsr:2,build37:10,buildver:10,bunch:9,bz2:10,calcul:7,call:3,caller:[2,9],can:[1,7,10],care:2,categori:7,causal:7,cgatool:[],charact:6,check:[2,5],chr1:9,chr9:10,chromosom:9,cigar:2,clinvar_20170130:10,clone:4,cmat:7,cmc:7,column:0,com:4,combinevari:8,complet:10,conda:4,configurationfil:[2,8],contain:[5,8],content:3,convers:6,convert2annovar:10,convert:10,coordin:2,creat:[0,2,8],create_index:2,crr:10,csv:[0,6],csvout:10,cummul:7,current:2,cutadapt:1,cutadapt_path:1,cytoband:10,dat:5,data:[1,3,6,8],dbnsfp30a:10,dbsnp138:2,decreas:7,dedup:2,defin:6,detect:[2,8],differ:9,dir:6,direct:7,dlog4j:[2,8],dnb:10,don:[1,7],done:9,dontusesoftclippedbas:2,download:0,downregul:6,duplic:2,e00461_116:2,e00461_116_gw170602261_1:2,each:7,effect:7,enough:1,entrez:6,entrezid:6,etc:7,evid:[],evidence2sam:10,evidencednb:10,exac03:10,exampl:[7,10],exclud:9,exist:7,extend2:1,fall:7,fals:6,famou:7,fasta:[2,5,8],fastq:[1,2],fastqc:1,fastqc_arg:1,few:7,file1:9,file2:9,file3:9,file:[0,2,8,10],filter:6,filteredareuncal:8,filteredrecordsmergetyp:8,first:[2,8],foldchang:6,folder:10,follow:[1,2,7,8],format:5,found:10,fraction:7,frame:6,frequpp:7,from:[2,3,4,9,10],ftp:0,further:[2,10],gage:6,galor:[1,4],gatk:[2,8],gene:[5,6,7],gene_xref:10,genefil:7,geneid:0,gener:[2,3],genom:[1,3],genomeanalysistk:[2,8],genomedir:2,genomefastafil:2,genomegener:2,get:6,gff:0,git:4,github:4,gnomad_exom:10,gobpr:6,gobpset:6,good:1,gov:0,gpr98:7,grch37:0,grch37_latest:0,grch37_latest_genom:0,greater:6,group:[7,9],gs000004829:10,gs4829_9:10,gset:6,gspatientx:10,gtf:5,gzip:1,gzvcf:9,h_sapien:0,haplotyp:2,haplotypecal:2,has:7,have:[1,2,7],head:[0,6],here:[0,1,2,5,7,8,9,10],hg19:10,hg38:[2,8],hg38index:2,hist_merg:1,histmap1mtxt:1,histmap4m:1,histogram:1,homo_sapiens_assembly38:2,how:2,howev:1,hsa:6,html:4,http:4,human_genom:1,human_hexam:5,human_train:5,humandb:10,idtyp:6,idx:6,igv:10,ihist:1,illumina:2,in1:1,in2:1,includ:7,increas:7,indel:2,index:[2,10],inform:7,input:10,instal:[1,3,5,7],instead:7,interest:[2,7],interv:2,invcf:7,its:1,jar:[2,8],java:[2,8],kbac:7,keep_if_any_unfilt:8,kegg:6,keggr:6,keggresidsdn:6,keggresidsup:6,keggrespathwaysdn:6,keggrespathwaysup:6,kei:0,kernel:7,keytyp:0,known:2,known_indel:2,knownsit:2,lappli:6,later:6,less:[6,7],let:[2,6,7],list:8,lncscore:5,log4j2:[2,8],logfc:6,lot:7,maf:7,makesitesonlyvcf:8,maketxdbfromgff:0,mani:7,mark:2,markdupl:2,megr:[],memori:1,merg:9,method:7,metric:2,million:1,mills_and_1000g_gold_standard:2,minn:8,mitimpact2:10,more:[1,7],mutat:[2,3],mutect2:2,mytranscript:5,myvcf:9,name:[6,8],namefold:6,nastr:10,natur:4,ncbi:0,necessari:2,need:[1,2,6,8,9],next:3,nih:0,nlm:0,noncod:3,normal1:8,normal2:8,normal3:8,normal:8,now:[2,10],one:9,onli:8,oper:[9,10],our:[2,5],ourlncrna:5,out:[2,7,9,10],outdir:1,output:[1,2,10],pair1:[1,2],pair2:[1,2],pair:1,pair_1:1,pair_2:1,panel:8,paramet:7,pass:[2,9],path_to_cutadapt:1,pathview:6,patientx:10,pdf:2,ped:7,per:[7,9],perced:8,percentag:7,pheno:7,phenotyp:7,picard:[2,8],pid:6,pipelin:[2,8],plot:[2,6],plot_pathwai:6,polish:10,pon:8,pon_siteonli:8,post:2,post_recal_data:2,power:7,pre:2,predict:3,prefer:7,prefilt:1,pretti:1,printread:2,prior:7,project:4,protocol:10,python:5,qualiti:1,queri:9,rare:3,read:2,readfilesin:2,readthedoc:4,realign:2,realignertargetcr:2,reassignonemappingqu:2,recablir:2,recal_data:2,recalibration_plot:2,recod:9,ref:[1,9],refer:[1,2,10],refflat_hg19:7,refgen:10,refseq:0,refseq_identifi:0,regardless:7,region:7,regulari:9,rem:1,remov:[1,9,10],requir:1,res:6,result:5,retain_unpair:1,revel:10,rg_added_sort:2,rgid:2,rglb:2,rgpl:2,rgpu:2,rgsm:2,risk:7,rmqf:2,rmqt:2,rna:3,rnaseq:3,robust:7,row_numb:6,rownam:6,run:7,runmod:2,runthreadn:2,rvtest:7,s1_1:2,s1_2:2,sam:2,same:[6,7],sampl:[2,7,9],samtool:10,sappli:6,select:0,sequenc:3,set:[6,7],setkei:8,sherin:3,show:10,sigmet:6,signatur:6,silent:2,singl:7,site:8,sjdbfilechrstartend:2,sjdboverhang:2,skat:7,skato:7,small:7,somat:[2,3],some:[2,6],sort:10,sp1483:2,sp1483_s13:2,speci:6,specif:7,specifi:7,split:2,splitncigarread:2,srep34838:4,stabl:4,stand_call_conf:2,star:2,start:[6,7],step:2,stop:6,studi:7,sub:6,subset:9,substr:6,summar:7,tab:2,tabix:9,tabl:2,table_annovar:10,tar:4,tbl_df:6,than:[2,7],thi:[2,7],thread:2,tip:5,tmpdn:6,tmpup:6,togeth:7,toolkit:3,top:6,trait:7,transcript:5,trim1:1,trim:4,trim_galor:[1,4],trimgalor:1,tsv:5,tutori:7,txname:0,txt:[1,7,10],type:7,under:7,upregul:6,use:[1,2,5,9,10],used:[2,10],ush2a:7,using:[1,7],validation_string:2,variant:[3,9],variou:10,vcf:[2,3,7,8],vcftool:9,view:10,weight:7,wget:[0,4],wglab:4,when:7,where:[2,10],whether:[2,7],which:[2,5],whole:3,write:[0,6],written:1,www:4,xml:[2,8],xmx50g:1,xref:10,xzvf:4,you:[1,2,7],your:0,zhanxw:4},titles:["**Annotation using tx2gene and txdb **","<strong>Next Generation Sequencing Toolkit</strong>","<strong>Variant Calling for RNASeq Data</strong>","Welcome to BitByBit\u2019s documentation!","<strong>Installation Assistance</strong>","<strong>Long nonCoding RNA Prediction</strong>","**Pathway Analysis for RNASeq and Microarrays **","<strong>Rare Variants Association Analysis</strong>","<strong>Somatic Mutations from RNAseq</strong>","<strong>VCF Toolkit</strong>","<strong>Whole Genomes Toolkit</strong>"],titleterms:{"long":5,adapt:7,analysi:[6,7],annot:0,annovar:10,assist:4,associ:7,bitbybit:3,burden:7,call:2,cgatool:10,combin:7,compon:7,cutadapt:4,data:2,document:3,estim:1,evid:10,fastqc:4,file:9,filter:9,format:10,from:8,gener:1,genom:10,insert:1,instal:4,lncscore:4,megr:9,microarrai:6,mutat:8,next:1,noncod:5,pathwai:6,predict:5,rare:7,read:1,rna:5,rnaseq:[2,6,8],rvtest:4,sam:10,sequenc:1,size:1,somat:8,sort:9,split:9,test:7,toolkit:[1,9,10],trim:1,trimgalor:4,tsv:10,tx2gene:0,txdb:0,using:0,varianc:7,variant:[2,7],vcf:9,welcom:3,whole:10}})