

<!doctype html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
        <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Variant Calling &#8212; BitByBit  documentation</title>
    <link rel="stylesheet" href="_static/bizstyle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/bizstyle.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="VCF Toolkit" href="vcfrandom.html" />
    <link rel="prev" title="Next Generation Sequencing Toolkit" href="NGS.html" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <!--[if lt IE 9]>
    <script type="text/javascript" src="_static/css3-mediaqueries.js"></script>
    <![endif]-->
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="vcfrandom.html" title="VCF Toolkit"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="NGS.html" title="Next Generation Sequencing Toolkit"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#"><strong>Variant Calling</strong></a><ul>
<li><a class="reference internal" href="#in-progress">In Progress</a></li>
<li><a class="reference internal" href="#calling-variants-using-samtools">Calling variants using Samtools</a></li>
<li><a class="reference internal" href="#calling-variants-using-gatk-pipeline">Calling variants using GATK pipeline</a></li>
<li><a class="reference internal" href="#annovar-annotations">Annovar Annotations</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="NGS.html"
                        title="previous chapter"><strong>Next Generation Sequencing Toolkit</strong></a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="vcfrandom.html"
                        title="next chapter"><strong>VCF Toolkit</strong></a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/variantcalling.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="variant-calling">
<h1><strong>Variant Calling</strong><a class="headerlink" href="#variant-calling" title="Permalink to this headline">¶</a></h1>
<div class="section" id="in-progress">
<h2>In Progress<a class="headerlink" href="#in-progress" title="Permalink to this headline">¶</a></h2>
<p>Here we will follow do variant calling, note that we can follow GATK pipeline or use samtools, we will show both.</p>
<p>Lets trim our data, we can refer also to our section  <span class="xref std std-doc">trimming</span> for more details</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">trim_galore</span> <span class="o">--</span><span class="n">paired</span> <span class="o">--</span><span class="n">gzip</span> <span class="o">--</span><span class="n">retain_unpaired</span> <span class="o">--</span><span class="n">trim1</span> <span class="o">--</span><span class="n">fastqc</span> <span class="o">--</span><span class="n">fastqc_args</span> <span class="s2">&quot;--outdir fastqc&quot;</span> <span class="o">-</span><span class="n">o</span> <span class="n">galore</span> <span class="o">--</span><span class="n">path_to_cutadapt</span> <span class="n">cutadapt</span><span class="o">/</span> \
<span class="n">reads_mate1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="n">read_mate2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Then, lets use bwa aln to align our reads</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bwa</span> <span class="n">aln</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span>  <span class="n">read_mate1_val_1</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>  <span class="o">&gt;</span> <span class="n">read_mate1</span><span class="o">.</span><span class="n">sai</span>
<span class="n">bwa</span> <span class="n">aln</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span>  <span class="n">read_mate2_val_2</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>  <span class="o">&gt;</span> <span class="n">read_mate2</span><span class="o">.</span><span class="n">sai</span>
</pre></div>
</div>
<p>Convert to sam:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bwa</span> <span class="n">sampe</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="n">read_mate1</span><span class="o">.</span><span class="n">sai</span> <span class="n">read_mate2</span><span class="o">.</span><span class="n">sai</span> <span class="n">galore</span><span class="o">/</span><span class="n">read_mate1_val_1</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="n">galore</span><span class="o">/</span><span class="n">read_mate2_val_2</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="n">reads</span><span class="o">.</span><span class="n">sam</span>
</pre></div>
</div>
<p>And convert our sam to bam using a sorted reference genome genome.fa.fai</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="kn">import</span> <span class="nn">genome.fa.fai</span> <span class="n">reads</span><span class="o">.</span><span class="n">sam</span> <span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Finally sort the bam files</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">T</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="n">reads</span><span class="o">.</span><span class="n">bam</span>
<span class="n">samtools</span> <span class="n">index</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</div>
<div class="section" id="calling-variants-using-samtools">
<span id="calling-samtools"></span><h2>Calling variants using Samtools<a class="headerlink" href="#calling-variants-using-samtools" title="Permalink to this headline">¶</a></h2>
<p>Here, we will call variants using samtools. We can also use the sorted BAM to call variants using Gatk pipeline <span class="xref std std-doc">calling_gatk</span></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">mpileup</span> <span class="o">-</span><span class="n">uf</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">|</span> <span class="n">bcftools</span> <span class="n">call</span> <span class="o">-</span><span class="n">vmO</span> <span class="n">v</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">vcf</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">2</span>
</pre></div>
</div>
</div>
<div class="section" id="calling-variants-using-gatk-pipeline">
<span id="calling-gatk"></span><h2>Calling variants using GATK pipeline<a class="headerlink" href="#calling-variants-using-gatk-pipeline" title="Permalink to this headline">¶</a></h2>
<p>Lets add read groups to our sam file reads.sam</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">Dlog4j</span><span class="o">.</span><span class="n">configurationFile</span><span class="o">=</span><span class="s2">&quot;log4j2.xml&quot;</span> <span class="o">-</span><span class="n">jar</span> <span class="n">picard</span><span class="o">.</span><span class="n">jar</span> <span class="n">AddOrReplaceReadGroups</span> <span class="n">I</span><span class="o">=</span><span class="n">reads</span><span class="o">.</span><span class="n">sam</span> <span class="n">O</span><span class="o">=</span><span class="n">rg_added_sorted</span><span class="o">.</span><span class="n">bam</span> \
 <span class="n">SO</span><span class="o">=</span><span class="n">coordinate</span> <span class="n">RGID</span><span class="o">=</span><span class="nd">@HWI</span><span class="o">-</span><span class="n">D00380_37_C4H2JACXX_2</span> <span class="n">RGLB</span><span class="o">=</span><span class="n">D109573_TAAGGCGA</span> <span class="n">RGPL</span><span class="o">=</span><span class="n">ILLUMINA</span> <span class="n">RGPU</span><span class="o">=</span><span class="nd">@HWI</span><span class="o">-</span><span class="n">D00380_37_C4H2JACXX_2</span><span class="o">.</span><span class="n">D109573_TAAGGCGA</span> <span class="n">RGSM</span><span class="o">=</span><span class="n">D109573</span>
</pre></div>
</div>
<p>Then lets do some sorting</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">sort</span> <span class="o">-</span><span class="n">T</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> <span class="n">rg_added_sorted</span><span class="o">.</span><span class="n">bam</span>
<span class="n">samtools</span> <span class="n">index</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Indel realignments is optional since we will use haplotype caller.
So lets do base recabliration</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">BaseRecalibrator</span> <span class="o">-</span><span class="n">R</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="o">-</span><span class="n">I</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">-</span><span class="n">knownSites</span> <span class="n">dbsnp_138</span><span class="o">.</span><span class="n">hg19</span><span class="o">.</span><span class="n">vcf</span>  <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">recal_data</span><span class="o">.</span><span class="n">table</span>
</pre></div>
</div>
<p>And post recabliration</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">BaseRecalibrator</span> <span class="o">-</span><span class="n">R</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="o">-</span><span class="n">I</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">-</span><span class="n">knownSites</span> <span class="n">dbsnp_138</span><span class="o">.</span><span class="n">hg19</span><span class="o">.</span><span class="n">vcf</span> <span class="o">-</span><span class="n">BQSR</span> <span class="n">reads</span><span class="o">.</span><span class="n">recal_data</span><span class="o">.</span><span class="n">table</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">post_recal_data</span><span class="o">.</span><span class="n">table</span>
</pre></div>
</div>
<p>Then get our recablirated bam file</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">PrintReads</span> <span class="o">-</span><span class="n">R</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="o">-</span><span class="n">I</span> <span class="n">reads</span><span class="o">.</span><span class="n">sorted</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">-</span><span class="n">BQSR</span> <span class="n">reads</span><span class="o">.</span><span class="n">post_recal_data</span><span class="o">.</span><span class="n">table</span> <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">recablirated</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Finally, lets use haplotype caller to call variants</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">java</span> <span class="o">-</span><span class="n">jar</span> <span class="n">GenomeAnalysisTK</span><span class="o">.</span><span class="n">jar</span> <span class="o">-</span><span class="n">T</span> <span class="n">HaplotypeCaller</span> <span class="o">-</span><span class="n">R</span> <span class="n">genome</span><span class="o">.</span><span class="n">fa</span> <span class="o">-</span><span class="n">I</span> <span class="n">reads</span><span class="o">.</span><span class="n">recablirated</span><span class="o">.</span><span class="n">bam</span> \
<span class="o">-</span><span class="n">dontUseSoftClippedBases</span> <span class="o">-</span><span class="n">stand_call_conf</span> <span class="mf">20.0</span>  <span class="o">-</span><span class="n">o</span> <span class="n">reads</span><span class="o">.</span><span class="n">haplotype</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</div>
<div class="section" id="annovar-annotations">
<h2>Annovar Annotations<a class="headerlink" href="#annovar-annotations" title="Permalink to this headline">¶</a></h2>
<p>Lets annotate our vcf file using Annovar</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">table_annovar</span><span class="o">.</span><span class="n">pl</span> <span class="n">reads</span><span class="o">.</span><span class="n">haplotype</span><span class="o">.</span><span class="n">vcf</span> <span class="n">humandb</span><span class="o">/</span> <span class="o">-</span><span class="n">buildver</span> <span class="n">hg19</span> <span class="o">-</span><span class="n">out</span> <span class="n">annotated_vcf</span> <span class="o">-</span><span class="n">remove</span> <span class="o">-</span><span class="n">protocol</span> <span class="n">refGene</span><span class="p">,</span><span class="n">ensGene</span><span class="p">,</span><span class="n">cytoBand</span><span class="p">,</span><span class="n">exac03</span><span class="p">,</span><span class="n">gnomad_exome</span><span class="p">,</span><span class="n">avsnp147</span><span class="p">,</span>\
<span class="n">dbnsfp30a</span><span class="p">,</span><span class="n">clinvar_20170130</span><span class="p">,</span><span class="n">mitimpact2</span><span class="p">,</span><span class="n">revel</span> <span class="o">-</span><span class="n">operation</span> <span class="n">g</span><span class="p">,</span><span class="n">g</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span><span class="p">,</span><span class="n">f</span>  <span class="o">-</span><span class="n">nastring</span> <span class="o">.</span> <span class="o">-</span><span class="n">vcfinput</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="vcfrandom.html" title="VCF Toolkit"
             >next</a> |</li>
        <li class="right" >
          <a href="NGS.html" title="Next Generation Sequencing Toolkit"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Sherine Awad.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.0b1.
    </div>
  </body>
</html>