<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Whole Genomes Analysis &#8212; BitByBit  documentation</title>
    
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Prediction of Long nonCoding RNA Prediction" href="lncRNA.html" />
    <link rel="prev" title="Rare Variants Association Analysis" href="rvassocanalysis.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="lncRNA.html" title="Prediction of Long nonCoding RNA Prediction"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="rvassocanalysis.html" title="Rare Variants Association Analysis"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="rvassocanalysis.html"
                        title="previous chapter"><strong>Rare Variants Association Analysis</strong></a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="lncRNA.html"
                        title="next chapter">Prediction of Long nonCoding RNA Prediction</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/wholegenomes.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="whole-genomes-analysis">
<h1>Whole Genomes Analysis<a class="headerlink" href="#whole-genomes-analysis" title="Permalink to this headline">Â¶</a></h1>
<p>Here, we will show various operations in whole genome Complete Genomics or BGI.</p>
<p>Convert to bam from cgatools, and then we can view the sorted bam in IGV:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>${CGATOOLS}/cgatools evidence2sam \
--evidence-dnbs=evidenceDnbs-chr9-GS000004829-ASM.tsv.bz2 \
--beta \
--reference=build37.crr | samtools view -uS - | \
samtools sort -T gs4829_9.sorted -o chr9-GS000004829.sorted.bam &amp;&amp; samtools index chr9-GS000004829.sorted.bam
</pre></div>
</div>
<p>Here we used the evidence file from cgatools output, and used evidence2sam to convert it to sam then sorted bam.</p>
<p>Convert from cgatools .tsv to Annovar format to further annotate with Annovar:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>${ANNOVAR}/convert2annovar.pl -format cg -out patientX.annovar.input  var-GSPatientX-ASM.tsv
</pre></div>
</div>
<p>Now, we can use patientX.annovar.input in Annovar:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>${ANNOVAR}/table_annovar.pl patientX.annovar.input  ${ANNOVAR}/humandb/ -buildver hg19 -out patientX -remove -protocol refGene,cytoBand,gnomad_exome,exac03,avsnp147,\
   dbnsfp30a,clinvar_20170130,mitimpact2,revel -operation g,f,f,f,f,f,f,f,f  -nastring . -csvout -polish -xref ${ANNOVAR}/example/gene_xref.txt
</pre></div>
</div>
<p>where example/gene_xref.txt is the xref file found in example folder of Annovar.</p>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="lncRNA.html" title="Prediction of Long nonCoding RNA Prediction"
             >next</a> |</li>
        <li class="right" >
          <a href="rvassocanalysis.html" title="Rare Variants Association Analysis"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">BitByBit  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Sherine Awad.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.5.3+.
    </div>
  </body>
</html>